learned.lic

```diff
@@ -4,27 +4,105 @@
+    armor_skills = ["Shield Usage", "Chain Armor", "Plate Armor", "Light Armor", "Brigandine", "Defending"]
+    weapon_skills = ["Parry Ability", "Small Edged", "Larged Edged", "Twohanded Edged", "Small Blunt", "Large Blunt", "Twohanded Blunt", "Slings", "Bow", "Crossbow", "Staves", "Polearms", "Light Thrown", "Heavy Thrown", "Brawling", "Offhand Weapon", "Melee Mastery", "Missile Mastery"]
+    magic_skills = ["Arcane Magic", "Holy Magic", "Life Magic", "Elemental Magic", "Lunar Magic", "Attunement", "Arcana", "Targeted Magic", "Inner Fire", "Inner Magic", "Augmentation", "Debilitation", "Utility", "Warding", "Sorcery"]
+    guild_skills = ["Empathy", "Astrology", "Expertise", "Scouting", "Backstab", "Summoning", "Bardic Lore" "Conviction", "Theurgy", "Thanatology", "Trading"]
+
+    if args.survival
+      @skills_to_show.concat(survival_skills)
+    elsif args.combats
+      @skills_to_show.concat(armor_skills)
+                     .concat(weapon_skills)
+                     .push('Targeted Magic')
+                     .push('Debilitation')
+                     .push('Evasion')
+                     .push('Tactics')
+    elsif args.magic
+      @skills_to_show.concat(magic_skills)
+    elsif args.armor
+      @skills_to_show.concat(armor_skills)
+    elsif args.weapons
+      @skills_to_show.concat(weapon_skills)
+    elsif args.lore
+      @skills_to_show.concat(lore_skills)
+    elsif args.guild
+      @skills_to_show.concat(guild_skills)
+    else
+      @skills_to_show.concat(survival_skills)
+                     .concat(lore_skills)
+                     .concat(armor_skills)
+                     .concat(weapon_skills)
+                     .concat(magic_skills)
+                     .concat(guild_skills)
+    end
+  end
+
+  pause
 
-total = DRSkill.list.select { |item| item.current - item.baseline > 0 }.map { |skill| DRSkill.gained_exp(skill.name) }.inject(&:+).round(2)
-tdps = DRSkill.list.select { |item| item.current - item.baseline > 0 }.map { |skill| (DRSkill.gained_exp(skill.name) * DRSkill.getrank(skill.name)) / 200 }.inject(&:+).round(2)
+  def format_skill_data(skill, time)
+    gain = DRSkill.gained_exp(skill.name)
+    format("%s %s (%0.2f\/hr\, %0.2f\/day)", skill.name.center(18), gain.to_s.center(6), gain / time, gain / time * 24)
+  end
 
-per_hour = total / learning_time.round(2)
-per_day  = per_hour * 24
+  def display_data
+    learning_time = ((Time.now - DRSkill.start_time) / 60.0 / 60)
+    columns = get_settings.learned_column_count
 
-daily_tdps = tdps / learning_time * 24
+    DRSkill.list.select { |item| @skills_to_show.include?(item.name)}
+                .select { |item| item.current - item.baseline > 0 }
+                .sort_by { |item| item.current - item.baseline }
+                .reverse.each_slice(columns) do |skills|
+                  output = skills.map { |skill| format_skill_data(skill, learning_time) }.join
```

Comment:
Note to self: I think this `.map` can be pulled out of the `.each_slice`. Outside the scope of this PR, I'll test it after we merge this.